src/installer/corehost/cli/apphost/static/CMakeLists.txt

```diff
@@ -30,6 +30,8 @@ set(HEADERS
     ../../../hostfxr_resolver.h
 )
 
+list(APPEND SOURCES $<TARGET_OBJECTS:libhostfxr_static> $<TARGET_OBJECTS:libhostpolicy_static>)
```

Comment:
I wonder why was this change needed.  Linking to the static libraries should work fine. The only case when we need to use object libraries is when a static library is built as a combination of other libraries, but it is not the case here.

Reply:
The `libhostfxr_static` and `libhostpolicy_static` targets are object libraries. The corehost CMake infra could really use some updates and cleanup to be more straightforward. I'd like to do that in a future PR.

Reply:
Hmm, interesting. I can see that they were changed to object ones few months ago and I am not sure why.

I agree that the corehost cmakefiles are very difficult to mentally parse. They looks as if msbuild style of build programming was used here. Instead of using functions for doing common stuff, ambient cmake variables are being set and then common helper cmake files included. I'd love to see it transformed to a natural cmake way at some point.